server/src/main/java/com/distkv/server/storeserver/runtime/heartbeat/HeartbeatManager.java

```diff
@@ -0,0 +1,53 @@
+package com.distkv.server.storeserver.runtime.heartbeat;
+
+import com.distkv.common.NodeInfo;
+import com.distkv.server.metaserver.client.DmetaClient;
+import com.distkv.server.metaserver.server.bean.HeartBeatResponse;
+import com.distkv.server.storeserver.runtime.slave.SlaveClient;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class HeartbeatManager {
+
+  private static ScheduledExecutorService scheduledExecutor = Executors
+      .newSingleThreadScheduledExecutor();
+
+  private static DmetaClient dmetaClient;
+
+  private static HashSet<String> buildSet;
+
+  private static AtomicInteger i = new AtomicInteger(0);
+
+  public HeartbeatManager(NodeInfo nodeInfo, int heartBeatInterval,
+                          String dmetaServerListStr, CopyOnWriteArrayList<SlaveClient> clients) {
+    buildSet = new HashSet<>();
+    buildSet.add(nodeInfo.getNodeName());
+    dmetaClient = new DmetaClient(dmetaServerListStr);
+    scheduledExecutor.scheduleAtFixedRate(() -> {
+      HeartBeatResponse response = dmetaClient.heartBeat(nodeInfo);
+      if (response.getNodeTable().size() > buildSet.size()) {
```

Comment:
How about adding a `diff()` method for `AbstractView`？ so that we have the ability to diff 2 views like:
```java
boolean is_different = view1.diff(view2);
if (is_different) {
     view1.loadsFrom(view2);
}
```
Add the implementation of `diff()` could be very simple: check the md5 code of the 2 views.

Reply:
What we are doing now is the heart processing. The view diff should be processed when distkv client pulls the global view.